File: languages/java/tests/src/test/java/co/revaly/sdk/TestClient.java
package co.revaly.sdk;

import co.revaly.sdk.core.model.PaymentRequest;
import co.revaly.sdk.testing.RapMockTransport;
import java.time.Duration;

/** Shared test wiring: a mock-backed client with synthetic-only configuration. */
final class TestClient {

    static final String API_KEY = "test-key-synthetic";
    static final String MTX = "mtx-synthetic-1";

    private TestClient() {}

    static RapClient client(RapMockTransport mock) {
        return client(mock, "2.1");
    }

    static RapClient client(RapMockTransport mock, String apiVersion) {
        return RapClient.builder()
                .apiKey(API_KEY)
                .baseUrl("https://sandbox.synthetic.test")
                .apiVersion(apiVersion)
                .overallDeadline(Duration.ofSeconds(5))
                .transport(mock)
                .build();
    }

    static PaymentRequest paymentRequest() {
        return new PaymentRequest().merchantTransactionId(MTX);
    }
}

File: languages/dotnet/tests/Revaly.Sdk.Tests/TestSupport/TestClient.cs
using Microsoft.Extensions.Logging;
using Revaly.Sdk.Core.Client;
using Revaly.Sdk.Core.Model;
using Revaly.Sdk.Logging;
using Revaly.Sdk.Testing;

namespace Revaly.Sdk.Tests.TestSupport;

/// <summary>Builders shared across the suite — every client here runs on the mock transport.</summary>
public static class TestClient
{
    public static RapClient Create(
        RapMockTransport mock,
        string apiVersion = "2.1",
        ILoggerFactory? loggerFactory = null,
        Action<RapWireTrace>? wireTraceHook = null,
        TimeSpan? overallDeadline = null)
    {
        return new RapClient(new RapClientOptions
        {
            ApiKey = SyntheticData.TestApiKey,
            BaseUrl = new Uri("https://sandbox.synthetic.test"),
            ApiVersion = apiVersion,
            LoggerFactory = loggerFactory,
            WireTraceHook = wireTraceHook,
            Transport = mock,
            OverallDeadline = overallDeadline,
        });
    }

    /// <summary>
    /// A minimal valid charge request carrying the synthetic test card.
    /// paymentMethodType is set explicitly: the generated PaymentRequest serializer
    /// dereferences the optional enum without an IsSet guard (core defect, tracked for a
    /// pipeline template-fork fix), so omitting it crashes before the wire.
    /// </summary>
    public static PaymentRequest ChargeRequest(string merchantTransactionId = SyntheticData.MerchantTransactionId)
        => new(
            amount: 1999,
            merchantTransactionId: merchantTransactionId,
            paymentMethodType: new Option<PaymentRequest.PaymentMethodTypeEnum?>(
                PaymentRequest.PaymentMethodTypeEnum.CreditCard),
            currency: new Option<string?>("USD"),
            paymentMethod: new Option<PaymentMethod?>(new PaymentMethod(
                creditCard: new Option<CreditCard?>(new CreditCard(
                    SyntheticData.TestPan, "12", "2030",
                    cardVerificationCode: new Option<string?>(SyntheticData.TestCvv),
                    cardType: new Option<CreditCard.CardTypeEnum?>(CreditCard.CardTypeEnum.Visa))))));
}
